app/assets/javascripts/directives/views/actionsMenu.ts

```diff
@@ -56,16 +56,17 @@ class ActionsMenuCtrl extends PureViewCtrl implements ActionsMenuScope {
     const actionExtensions = this.application.actionsManager!.getExtensions().sort((a, b) => {
       return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1;
     });
-    const extensionsForItem = await Promise.all(actionExtensions.map((extension) => {
-      return this.application.actionsManager!.loadExtensionInContextOfItem(
+    await Promise.all(actionExtensions.map(async (extension) => {
+      const extensionsForItem = await this.application.actionsManager!.loadExtensionInContextOfItem(
         extension,
         this.props.item
       );
+      const extensions = this.state.extensions || [];
+      this.setState({
+        extensions: extensions.concat(extensionsForItem)
+      });
     }));
     this.loadingExtensions = false;
```

Comment:
I think loadingExtensions should not be global, but per extension: https://github.com/standardnotes/web/blob/d192594aebcda9a00b7f320000e72917d4df8f37/app/assets/templates/directives/actions-menu.pug#L10

Reply:
`this.loadingExtensions` is used to show a spinner while extensions are loaded.

Reply:
Yes but every extension should have its own loading spinner since they are loaded and displayed individually. Here you are displaying 1 single loading spinner for all extensions, which doesn't make as much sense.